nibabel/cmdline/diff.py

```diff
@@ -313,7 +320,8 @@ def main(args=None, out=None):
         files,
         header_fields=opts.header_fields,
         data_max_abs_diff=opts.data_max_abs_diff,
-        data_max_rel_diff=opts.data_max_rel_diff
+        data_max_rel_diff=opts.data_max_rel_diff,
+        dtype=opts.dtype
```

Comment:
wouldn't work as is.  In cmdline you will get a string, like `"float64"`, whenever functions expect actual type, like `np.float64`

Reply:
untested code is broken code :-P

Reply:
Codecov appears green for it though? And I tested it in cmdline for myself. Could you please specify the use case you think I'm missing?

Reply:
This is fine because whenever we get a `dtype_var`, we generally immediately turn around and run `dtype(dtype_var)`, which will take `np.float64`, or `'<f8'` or `'float64'`.

https://github.com/nipy/nibabel/blob/7877add916fb1aa9732d347c21b2c2f1b1d4be45/nibabel/dataobj_images.py#L340

Reply:
still would be nice to have a test which invokes `main` here providing then dtype as a string.

Reply:
Sounds reasonable to me.

Reply:
Assuming these pass, please check to make sure this meets with your standards. I checked with @yarikoptic -- is this what you had in mind?